### PR TITLE
istioctl/cmd/remove-from-mesh:Don't remove 'istio-proxy' if it is only container and don't remove used volumes

### PR DESCRIPTION
**Pull request by tanjunchen**
[ ] Configuration Infrastructure
[ ] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by tanjunchen** [comment]
ref https://github.com/istio/istio/issues/25860

File: istioctl/cmd/remove-from-mesh.go [hunk 1]
```diff
@@ -201,7 +202,14 @@ func unInjectSideCarFromDeployment(client kubernetes.Interface, deps []appsv1.De
 		depName := strings.Join([]string{dep.Name, dep.Namespace}, ".")
 		sidecarInjected := false
 		podSpec := dep.Spec.Template.Spec.DeepCopy()
-		for _, c := range podSpec.Containers {
+		containers := podSpec.Containers
+		// The sidecar is always named 'istio-proxy', but it isn't a proxy if there is only one container.
+		// (For example, The pod container of ingressgateway is named "istio-proxy", but it isn't a sidecar)
+		if len(containers) == 1 && containers[0].Name == proxyContainerName {
```
Comment (shamsher31) [inline istioctl/cmd/remove-from-mesh.go:208]: I think `istio-proxy` will not always be the first container. Can you refer to this PR.
https://github.com/istio/istio/pull/25887/files#diff-beb2c770345282d0846ccdbaab7341d964054df8a3abf696ddee34125b30f86fR193-R202

**Pull request closed without merging**